migrations/inputs_docker/migration.go

```diff
@@ -0,0 +1,188 @@
+package inputs_docker
+
+import (
+	"fmt"
+
+	"github.com/influxdata/toml"
+	"github.com/influxdata/toml/ast"
+
+	"github.com/influxdata/telegraf/migrations"
+)
+
+// Migration function to migrate multiple deprecated Docker options
+func migrate(tbl *ast.Table) ([]byte, string, error) {
+	// Decode the old data structure
+	var plugin map[string]interface{}
+	if err := toml.UnmarshalTable(tbl, &plugin); err != nil {
+		return nil, "", err
+	}
+
+	var applied bool
+
+	// 1. Migrate container_names -> container_name_include
+	if containerNamesValue, found := plugin["container_names"]; found {
+		applied = true
+
+		// Convert to []interface{} for easier handling
+		var containerNames []interface{}
+		switch v := containerNamesValue.(type) {
+		case []interface{}:
+			containerNames = v
+		case []string:
+			for _, name := range v {
+				containerNames = append(containerNames, name)
+			}
+		default:
+			return nil, "", fmt.Errorf("container_names value is not a slice: %T", containerNamesValue)
+		}
```

Comment:
How about

```suggestion
			var err error
			containerNames, err = migrations.AsStringSlice(containerNamesValue)
			if err != nil {
				return nil, "", fmt.Errorf("'container_names' option: %w", err)
			}
```

Same for the other options below.